api/app/wildfire_one.py

```diff
@@ -117,15 +117,44 @@ def use_wfwx():
     return using_wfwx
 
 
+def _create_redis():
+    return StrictRedis(host=config.get('REDIS_HOST'),
+                       port=config.get('REDIS_PORT', 6379),
+                       db=0,
+                       password=config.get('REDIS_PASSWORD'))
+
+
 async def _fetch_access_token(session: ClientSession) -> dict:
     """ Fetch an access token for WFWX Fireweather API
     """
     logger.debug('fetching access token...')
     password = config.get('WFWX_SECRET')
     user = config.get('WFWX_USER')
     auth_url = config.get('WFWX_AUTH_URL')
-    async with session.get(auth_url, auth=BasicAuth(login=user, password=password)) as response:
-        return await response.json()
+    cache = _create_redis()
+    key = f'{auth_url}?{urlencode({user: user, password: password})}'
+    try:
+        cached_json = cache.get(key)
+    except Exception as error:  # pylint: disable=broad-except
+        cached_json = None
+        logger.error(error)
+    if cached_json:
+        logger.info('redis cache hit %s', auth_url)
+        response_json = json.loads(cached_json.decode())
+    else:
+        logger.info('redis cache miss %s', auth_url)
+        async with session.get(auth_url, auth=BasicAuth(login=user, password=password)) as response:
+            response_json = await response.json()
+            try:
+                if response.status == 200:
+                    # We expire when the token expires, or 10 minutes, whichever is less.
+                    # NOTE: only caching for 10 minutes right now, since we aren't handling cases
+                    # where the token is invalidated.
+                    expires = min(response_json['expires_in'], 600)
```

Comment:
Can we put this 10 minute literal in our config? I'm afraid of us adding lots of timed eviction entries and not being able to audit them easily in the future.

Reply:
Sure thing.